### PR TITLE
Used DevToolsRequest in Extension background to send tabUrl.

### PR DESCRIPTION
**Pull request by pisimulation**

### CODE REVIEW

File: dwds/debug_extension/web/background.dart [hunk 1]
```diff
@@ -28,12 +29,20 @@ void main() {
...
-        var port = e.result.value;
-        startSseClient(port, currentTab);
+      sendCommand(Debuggee(tabId: currentTab.id), 'Debugger.enable',
+          EmptyParam(), allowInterop((e) {}));
+      sendCommand(
+          Debuggee(tabId: currentTab.id),
+          'Runtime.evaluate',
+          ExtensionPortParam(
+              expression:
+                  '[\$extensionPort, \$extensionHostname, \$dartAppId, \$dartAppInstanceId]',
```
Comment (pisimulation) [inline dwds/debug_extension/web/background.dart]: Getting all injected information needed.

File: dwds/debug_extension/web/background.dart [hunk 2]
```diff
@@ -47,8 +56,14 @@ void main() {
...
-Future<void> startSseClient(port, currentTab) async {
-  var client = SseClient('http://localhost:$port/test');
+Future<void> startSseClient(
+    hostname, port, appId, instanceId, currentTab) async {
+  var client = SseClient('http://$hostname:$port/test');
+  await client.onOpen.first;
+  client.sink.add(jsonEncode(serializers.serialize(DevToolsRequest((b) => b
+    ..appId = appId as String
+    ..instanceId = instanceId as String
+    ..tabUrl = currentTab.url as String))));
```
Comment (pisimulation) [inline dwds/debug_extension/web/background.dart]: The handshake: SseClient sends a DevToolsRequest containing tabUrl.

File: dwds/lib/src/servers/extension_debugger.dart [hunk 3]
```diff
@@ -36,6 +46,9 @@ class ExtensionDebugger implements WipDebugger {
         }
         _completers[message.id]
             .complete(WipResponse(encodedResult as Map<String, dynamic>));
+      } else if (message is DevToolsRequest) {
+        _tabUrl = message.tabUrl;
+        _devToolsRequestController.sink.add(message);
```
Comment (pisimulation) [inline dwds/lib/src/servers/extension_debugger.dart]: The listener will sit in DevHandler.

File: dwds/debug_extension/web/background.dart [hunk 4]
```diff
@@ -28,12 +29,20 @@ void main() {
...
+      sendCommand(Debuggee(tabId: currentTab.id), 'Debugger.enable',
+          EmptyParam(), allowInterop((e) {}));
+      sendCommand(
+          Debuggee(tabId: currentTab.id),
+          'Runtime.evaluate',
+          ExtensionPortParam(
+              expression:
+                  '[\$extensionPort, \$extensionHostname, \$dartAppId, \$dartAppInstanceId]',
+              returnByValue: true), allowInterop((e) {
+        var port = e.result.value[0];
```
Comment (grouma) [inline dwds/debug_extension/web/background.dart]: You'll want to do some error handling here if the values are not available. Maybe open a pop up saying this isn't a Dart application.

File: dwds/debug_extension/web/background.dart [hunk 5]
```diff
@@ -28,12 +29,27 @@ void main() {
...
+              expression:
+                  '[\$extensionPort, \$extensionHostname, \$dartAppId, \$dartAppInstanceId]',
+              returnByValue: true), allowInterop((e) {
+        String port, hostname, appId, instanceId;
+        try {
+          port = e.result.value[0] as String;
+          hostname = e.result.value[1] as String;
+          appId = e.result.value[2] as String;
+          instanceId = e.result.value[3] as String;
+        } catch (_) {
```
Comment (grouma) [inline dwds/debug_extension/web/background.dart]: Is there a better error we can catch here? Is there a way we can just ensure that these values are empty?

**Pull request merged**